File: Algorithms/String/43. Multiply Strings.py
class Solution(object):
    def multiply(self, num1, num2):
		l1, l2 = len(num1), len(num2)
		pos = [0] * (l1 + l2)
		for i in range(l1 - 1, -1, -1):
			for j in range(l2 - 1, -1, -1):
				Sum = pos[i + j + 1] + int(num1[i]) * int(num2[j])
				pos[i + j] += Sum // 10
				pos[i + j + 1] =  Sum % 10
		for inx in range(l1 + l2):
			if(pos[inx]):
				return ''.join([str(x) for x in pos[inx:]])
		return '0'

File: Algorithms/Math/43. Multiply Strings.java
class Solution {
    public String multiply(String num1, String num2) {
        if(num1.equals("0") || num2.equals("0")) return "0";
        
        StringBuilder firstNumber = new StringBuilder(num1);
        StringBuilder secondNumber = new StringBuilder(num2);
        
        firstNumber.reverse();
        secondNumber.reverse();
        
        int N = firstNumber.length() + secondNumber.length();
        StringBuilder answer = new StringBuilder();
        for(int i = 0; i < N; ++i) {
            answer.append(0);
        }
        
        for(int place1 = 0; place1 < firstNumber.length(); ++place1) {
            int digit1 = firstNumber.charAt(place1) - '0';
            for(int place2 = 0; place2 < secondNumber.length(); ++place2) {
                int digit2 = secondNumber.charAt(place2) - '0';
                int curPos = place1 + place2;
                int carry = answer.charAt(curPos) - '0';
                int mul = digit1 * digit2 + carry;
                
                answer.setCharAt(curPos, (char)(mul % 10 + '0'));
                int val = (answer.charAt(curPos + 1) - '0') + mul / 10;
                answer.setCharAt(curPos + 1, (char)(val + '0'));
            }
        }
        if(answer.charAt(answer.length() - 1) == '0') {
            answer.deleteCharAt(answer.length() - 1);
        }
        answer.reverse();
        return answer.toString();
    }
}
